csharp/src/Drivers/Apache/Impala/ImpalaConnection.cs

```diff
@@ -88,5 +595,65 @@ public override IArrowArrayStream GetTableTypes()
         internal override SchemaParser SchemaParser { get; } = new HiveServer2SchemaParser();
 
         internal override IArrowArrayStream NewReader<T>(T statement, Schema schema, CancellationToken cancellationToken = default) => new HiveServer2Reader(statement, schema, dataTypeConversion: DataTypeConversion, cancellationToken);
+
+        protected abstract void ValidateConnection();
+        protected abstract void ValidateAuthentication();
+        protected abstract void ValidateOptions();
+
+        internal abstract ImpalaServerType ServerType { get; }
+
+        internal struct TableInfo(string type)
+        {
+            public string Type { get; } = type;
+
+            public List<string> ColumnName { get; } = new();
+
+            public List<short> ColType { get; } = new();
+
+            public List<string> BaseTypeName { get; } = new();
+
+            public List<string> TypeName { get; } = new();
+
+            public List<short> Nullable { get; } = new();
+
+            public List<int?> Precision { get; } = new();
+
+            public List<short?> Scale { get; } = new();
+
+            public List<int> OrdinalPosition { get; } = new();
+
+            public List<string> ColumnDefault { get; } = new();
+
+            public List<string> IsNullable { get; } = new();
+
+            public List<bool> IsAutoIncrement { get; } = new();
+        }
+
+        internal class ImpalaInfoArrowStream : IArrowArrayStream
```

Comment:
This can be moved to parent (`HiveServer2InfoArrowStream`) and shared with Impala/Spark.